src/panels/lovelace/hui-root.ts

```diff
@@ -228,7 +228,21 @@ class HUIRoot extends LitElement {
                   >
                     ${this.lovelace!.config.views.map(
                       (view) => html`
-                        <paper-tab aria-label="${view.title}">
+                        <paper-tab
+                          aria-label="${view.title}"
+                          class="${classMap({
+                            "hide-tab": Boolean(
+                              !this._editMode &&
+                                view.show !== undefined &&
+                                ((Array.isArray(view.show) &&
+                                  !view.show.some(
+                                    (e) => e.user === this.hass!.user!.id
+                                  )) ||
+                                  (typeof view.show === "boolean" &&
+                                    view.show === false))
```

Comment:
`view.show === false` will only match booleans, the triple `=` also checks type, the `==` doesn't.